clustering and desaturation upgrades

zhut19 opened the pull request:
Here we propose several upgrades tailored for high energy peaks.
Detailed information is documented in [clustering upgrades](https://xe1t-wiki.lngs.infn.it/doku.php?id=xenon:xenon1t:sim:notes:tzhu:high_energy_peak_clustering_amendment:multipeakseparation) and [desaturation upgrades](https://xe1t-wiki.lngs.infn.it/doku.php?id=xenon:xenon1t:chiara:pax_desaturation).

tunnell commented:
Looks nice, especially the S2 tails work

tunnell commented on pax/plugins/peak_processing/BasicProperties.py:
```diff
@@ -86,10 +87,10 @@ def transform_event(self, event):
-                self.log.warning("Sum waveform of peak %d-%d (%0.2f pe area) sums to a nonpositive value... unusual!"
-                                 " Cannot align peak's sum waveform, storing zeros instead." % (peak.left,
+                self.log.debug("Sum waveform of peak %d %d-%d (%0.2f pe area) sums to a nonpositive value... unusual!"
+                                 " Cannot align peak's sum waveform, and will be deleted." % (peak_i, peak.left,
```
__[Flake8]__

[E127] continuation line over-indented for visual indent


<!-- Comment by [Sider](https://sider.review) -->

feigaodm commented on pax/config/XENON1T.ini:
```diff
@@ -333,6 +334,13 @@ min_height = 25 #pe
 # Minimum min(pheight eak left of loc min, height of peak right of loc min)/height of loc min
 min_ratio = 4
 
+# [LocalMinimumClustering.TailSeparation]
```
delete the lines that doesn't add information

feigaodm commented on pax/plugins/peak_processing/LocalMinimumClustering.py:
```diff
@@ -116,34 +116,77 @@ def split_peak(self, peak, split_points):
+    return good_minima
+
+
+class TailSeparation(LocalMinimumClustering):
```
Add some description on this function

zhut19 replied:
More description are added to the function, but a relative short one for the class here.

feigaodm commented on pax/plugins/signal_processing/DesaturatePulses.py:
```diff
@@ -27,71 +27,117 @@ def transform_event(self, event):
+                if len(turning_point) > 0:
+                    end = last_saturated + turning_point[0]
+
+                # Determine where to stop correction
```
clean the lines if we don't need to leave them their for reference

zhut19 commented:
Thanks @sanderbreur 
 * We are thinking about checking 'tail_cutoff_ratio' using simulated waveform once we can match the waveforms. Not sure how to check it with processing data, except looking at the energy resolution.

 * We are checking the 'large_after_pulsing_channels' list changes between sr0, sr1 and after sr1 with pmt database. We can come up a strategy after checking.

 * We ran into some bug with a peak that set to zeros, and to by pass it, we added those line to actually delete the peaks. And we think those ~0.01 pe peaks won't matter anyway.

 * Line 123 is deleted

 * Added a condition to exclude S1 from TailSeparation.

 * Will add int16 to float 64 to the DesaturatePulses class description

chiaracapelli commented:
@sanderbreur The list of 'large_after_pulsing_channels' is checked and updates with the PMT database. We are using the 4 datasets during SR1 in the config/pmt_afterpulses, so in principle we don't have to update the list if we use SR1 data. We can update it including SR0 if we want to use also that.

sanderbreur commented:
Chiara, cool, would be nice to have the list later also for sr0, else sr0
reprocessing will give problems.


On Tue, 17 Jul 2018, 22:43 Chiara Capelli, <notifications@github.com> wrote:

> @sanderbreur <https://github.com/sanderbreur> The list of
> 'large_after_pulsing_channels' is checked and updates with the PMT
> database. We are using the 4 datasets during SR1 in the
> config/pmt_afterpulses, so in principle we don't have to update the list if
> we use SR1 data. We can update it including SR0 if we want to use also that.
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/XENON1T/pax/pull/712#issuecomment-405721169>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AGNstTYW_l9pS5kXkh1Zf02FwfQjhAovks5uHkx0gaJpZM4VPlY0>
> .
>

chiaracapelli commented:
![unknown](https://user-images.githubusercontent.com/20901250/42913125-0fbbc430-8ac1-11e8-8d3e-66aaa40dd8e3.png)

We reprocessed one dataset and checked the s1s and s2s relative difference between the old processed and this new version in 1T FV. The data are mostly in the region 300-1500 keV.
